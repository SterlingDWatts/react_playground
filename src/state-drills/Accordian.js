import React, { Component } from 'react';

export default class Accordian extends Component {
    static defaultProps = {
        section: []
    }
    state = {
        currentButton: null,
    }
    handleButtonClick(index) {
        this.setState({ currentButton: index })
    }
    renderContent(index) {
        if (index === this.state.currentButton) {
            return (
                <p>{this.props.section[index].content}</p>
            )
        } else {
            return <div></div>
        }
    }
    renderListItems() {
        return this.props.section.map((section, index) => (
            <li key={index}>
                <button key={index} onClick={() => this.handleButtonClick(index)}>{section.title}</button>
                {this.renderContent(index)}
            </li>
        ))
    }
    render() {
        return (
            <ul>
                {this.renderListItems()}
            </ul>
        )
    }
}