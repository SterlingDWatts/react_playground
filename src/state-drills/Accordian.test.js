import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Accordian from './Accordian';

describe('Accordian Component', () => {
    const sections = [
        {
            title: 'Section 1',
            content: 'Lorem ipsum dolor sit amet consectetur adipisicing elit.',
        },
        {
            title: 'Section 2',
            content: 'Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!',
        },
        {
            title: 'Section 3',
            content: 'Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?',
        }
    ]
    it('renders without errors', () => {
        const div = document.createElement('div');
        ReactDOM.render(<Accordian />, div);
        ReactDOM.unmountComponentAtNode(div);
    })
    it('renders an empty ul given no sections', () => {
        const wrapper = shallow(<Accordian />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })
    it('renders no section as active by default', () => {
        const wrapper = shallow(<Accordian section={sections}/>)
        expect(toJson(wrapper)).toMatchSnapshot()
    })
    it('opens a section when the button is clicked', () => {
        const wrapper = shallow(<Accordian section={sections}/>)
        wrapper.find('button').at(0).simulate('click')
        expect(toJson(wrapper)).toMatchSnapshot()
    })
    it('only has the last the section open from the last button clicked', () => {
        const wrapper = shallow(<Accordian section={sections}/>)
        wrapper.find('button').at(0).simulate('click')
        wrapper.find('button').at(1).simulate('click')
        expect(toJson(wrapper)).toMatchSnapshot()
    })
})