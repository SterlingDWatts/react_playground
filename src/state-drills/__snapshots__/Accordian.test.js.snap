// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordian Component only has the last the section open from the last button clicked 1`] = `
<ul>
  <li
    key="0"
  >
    <button
      key="0"
      onClick={[Function]}
    >
      Section 1
    </button>
    <div />
  </li>
  <li
    key="1"
  >
    <button
      key="1"
      onClick={[Function]}
    >
      Section 2
    </button>
    <p>
      Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!
    </p>
  </li>
  <li
    key="2"
  >
    <button
      key="2"
      onClick={[Function]}
    >
      Section 3
    </button>
    <div />
  </li>
</ul>
`;

exports[`Accordian Component opens a section when the button is clicked 1`] = `
<ul>
  <li
    key="0"
  >
    <button
      key="0"
      onClick={[Function]}
    >
      Section 1
    </button>
    <p>
      Lorem ipsum dolor sit amet consectetur adipisicing elit.
    </p>
  </li>
  <li
    key="1"
  >
    <button
      key="1"
      onClick={[Function]}
    >
      Section 2
    </button>
    <div />
  </li>
  <li
    key="2"
  >
    <button
      key="2"
      onClick={[Function]}
    >
      Section 3
    </button>
    <div />
  </li>
</ul>
`;

exports[`Accordian Component renders an empty ul given no sections 1`] = `<ul />`;

exports[`Accordian Component renders no section as active by default 1`] = `
<ul>
  <li
    key="0"
  >
    <button
      key="0"
      onClick={[Function]}
    >
      Section 1
    </button>
    <div />
  </li>
  <li
    key="1"
  >
    <button
      key="1"
      onClick={[Function]}
    >
      Section 2
    </button>
    <div />
  </li>
  <li
    key="2"
  >
    <button
      key="2"
      onClick={[Function]}
    >
      Section 3
    </button>
    <div />
  </li>
</ul>
`;
